### DISCUSSION TITLE
`input in flex scanner failed`

### DISCUSSION THREAD

**Discussion by poperigby** [troglobit/inadyn#440] [Q&A] [answered]
I'm trying to setup Inadyn, but I'm getting this error when I start it, and then it fails:

```
inadyn[1]: In-a-dyn version 2.12.0-dev -- Dynamic DNS update client.
input in flex scanner failed
```

This is my configuration file:

```
period          = 300
user-agent      = Mozilla/5.0

provider cloudflare.com {
    username = haddock.cc
    password = REDACTED
    hostname = haddock.cc
    ttl    = 1
    proxied  = false
}
```

**Comment by troglobit** [comment]
That's very odd. The error comes from flex, which is the LALR parser used in libConfuse, which In-a-Dyn in turn use to parse configuration files.  I have never seen this before.

This is how it looks on my system, Linux Mint 21.2, based on Ubuntu 22.04.

```
~/src/inadyn(master)$ ./src/inadyn -f bar.conf -n -l debug -1
inadyn[102230]: In-a-dyn version 2.12.0-dev -- Dynamic DNS update client.
inadyn[102230]: Guessing DDNS plugin 'default@cloudflare.com' from 'cloudflare.com'
inadyn[102230]: Resolving hostname haddock.cc => IP# 71.212.102.237
inadyn[102230]: Get address for default@cloudflare.com
inadyn[102230]: Checking for IP# change, initiating HTTPS ...
inadyn[102230]: Checking for IP# change, connecting to 1.1.1.1([1.1.1.1]:443)
...
```

My guess is that there's something really wrong with the GNU Flex package on your system.  I'm on version 2.6.4-8build2 (Debian appends -8build2, which may include various local fixes and backports they carry).  I don't know what kind of system you are on, but I'd suggest reporting this upstream, to your distribution.

**Reply by poperigby** [reply]
Sorry, I forgot to mention that I'm actually running In-a-Dyn in a Podman container, with [this](https://hub.docker.com/r/troglobit/inadyn) image.

Podman can use Kubernetes YAML files, so I used a ConfigMap for my configuration file, so maybe that's the problem. I wouldn't think so though, because I checked in my configuration volume, and my `inadyn.con` is right there, mounted in `/etc`.

**Reply by troglobit** [reply]
I've now tested by using plain docker, which seems to work fine here:

```
~/src/inadyn(master)$ docker pull troglobit/inadyn
...
~/src/inadyn(master)$ docker run --rm -it -v "$PWD/bar.conf:/etc/inadyn.conf" troglobit/inadyn:latest -l debug -1 -n
inadyn[1]: In-a-dyn version 2.12.0-dev -- Dynamic DNS update client.
inadyn[1]: Guessing DDNS plugin 'default@cloudflare.com' from 'cloudflare.com'
inadyn[1]: Resolving hostname haddock.cc => IP# 71.212.102.237
inadyn[1]: Get address for default@cloudflare.com
inadyn[1]: Checking for IP# change, initiating HTTPS ...
inadyn[1]: Checking for IP# change, connecting to 1.1.1.1([1.1.1.1]:443)
...
```

I have never set up anything with podman and k8s yaml files, first time I've heard of ConfigMap.  Please try running In-a-Dyn in a more conventional way first to isolate the problem.

**Reply by sleepntsheep** [reply]
I am having the same problem in docker, running the exact same command
`docker run --rm -it -v "$PWD/bar.conf:/etc/inadyn.conf" troglobit/inadyn:latest -l debug -1 -n`

**Reply by Trufax** [reply]
i have the same issue running it on Docker the exact same command.

**Comment by soda3x** [comment]
Having this issue too, using Docker and the official troglobit/inadyn:latest image

**Reply by troglobit** [reply]
Could you please re-run with the command line options I demonstrated in my attempt (above) to reproduce?

**Comment by arsini71** [comment, marked as answer]
I also had this issue with troglobit/inadyn:latest image  (v2.12) with volume definition:
-v /etc:/etc/inadyn.conf    
--> failed with "input in flex scanner failed"

fixed with change to volume definition:

-v /etc:/etc

**Reply by troglobit** [reply]
Awesome, thank you! <3
